packages/client/src/components/context/EmbedProvider.svelte

```diff
@@ -0,0 +1,41 @@
+<script>
+  import Provider from "./Provider.svelte"
+  import { onMount } from "svelte"
+
+  let data = {}
+
+  export function extractDomainFromUrl(url) {
+    const { hostname } = new URL(url)
+    const parts = hostname.split('.');
+    const tld = parts.slice(-2).join(".")
+    return tld
+  }
+
+  export function handleMessage(event) {
+    // Validate the event origin to ensure it's coming from a trusted source
+    // Allow different subdomains but must match TLD
+    const appOrigin = extractDomainFromUrl(window.location.origin)
+    const eventOrigin = extractDomainFromUrl(event.origin)
+
+    if (appOrigin !== eventOrigin) {
```

Comment:
I think this logic is inverted - it should be `===` here.

Reply:
Ah, good catch. Was messing around for testing